### ISSUE TITLE
generalize R.any

### ISSUE THREAD

**Issue by davidchambers** [ramda/ramda#2671]
Current type:

```haskell
R.any :: (a -> Boolean) -> Array a -> Boolean
```

Desired type:

```haskell
R.any :: Foldable f => (a -> Boolean) -> f a -> Boolean
```

**Comment by CrossEye** [comment]
I can certainly see a benefit.  Do you have an implementation in mind?

**Comment by davidchambers** [comment]
`S.reduce (S.or) (false)` is what I used instead.

**Comment by GingerPlusPlus** [comment]
How operating on iterables (but not iterators) instead of foldables to preserve short-circuiting?

**Comment by CrossEye** [comment]
@GingerPlusPlus: Operating on iterables seems a great idea, but it should probably be thought of as a library-wide enhancement, and not specific to one function.

**Comment by GingerPlusPlus** [comment]
Of course, `all` and `none` should be kept in sync with `any`. However, iterable producers, consumers (this and #2756) and transformations (#2585) are distinct and therefore should be discussed separately.

Is supporting foldables that are not iterables worth giving up easy short-circuiting?

**Comment by GingerPlusPlus** [comment]
The only foldable but not iterable types I'm aware of are Observables and Streams. Because of their async nature, they must wrap the result of reduction (complicating signatures of `any`, `all`, and `none`). Is it allowed by FantasyLand?
